### PR TITLE
Forward compatibility with jenkinsci/credentials-plugin#551

### PR DESCRIPTION
**Pull request by janfaracik**
https://github.com/jenkinsci/credentials-plugin/pull/551

### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by timja** [comment]
(can ignore JiraPluginTest failing)

**Comment by basil** [comment]
> when updating ATH tests for plugin changes, do we need to maintain compatibility with the old version of the plugin?

@janfaracik I think that is desirable to preserve compatibility unless it is prohibitively difficult. This PR passed against https://github.com/jenkinsci/credentials-plugin/pull/551 but is failing on the current release of the Credentials plugin with

```
java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
	at java.base/java.util.Objects.checkIndex(Objects.java:361)
	at java.base/java.util.ArrayList.get(ArrayList.java:427)
	at org.jenkinsci.test.acceptance.plugins.ssh_slaves.SshSlaveLauncher.addCredential(SshSlaveLauncher.java:39)
	at org.jenkinsci.test.acceptance.plugins.ssh_slaves.SshSlaveLauncher.pwdCredentials(SshSlaveLauncher.java:69)
```

In this case I think it would be relatively easy to support both cases with some sort of if/else or try/catch statement, so I think we should strive to preserve compatibility. See #1690 and #1693 for plenty of examples of how this was done before.

**Comment by janfaracik** [comment]
> > when updating ATH tests for plugin changes, do we need to maintain compatibility with the old version of the plugin?
> 
> @janfaracik I think it is desirable to preserve compatibility unless it is prohibitively difficult. This PR passed against [jenkinsci/credentials-plugin#551](https://github.com/jenkinsci/credentials-plugin/pull/551) but is failing on the current release of the Credentials plugin with
> 
> ```
> java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 0
> 	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
> 	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
> 	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
> 	at java.base/java.util.Objects.checkIndex(Objects.java:361)
> 	at java.base/java.util.ArrayList.get(ArrayList.java:427)
> 	at org.jenkinsci.test.acceptance.plugins.ssh_slaves.SshSlaveLauncher.addCredential(SshSlaveLauncher.java:39)
> 	at org.jenkinsci.test.acceptance.plugins.ssh_slaves.SshSlaveLauncher.pwdCredentials(SshSlaveLauncher.java:69)
> ```
> 
> In this case I think it would be relatively easy to support both cases with some sort of if/else or try/catch statement, so I think we should strive to preserve compatibility. See #1690 and #1693 for plenty of examples of how this was done before.

Sure thing, updated.

File: src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java [hunk 1]
```diff
@@ -33,14 +34,22 @@ public SshSlaveLauncher(PageObject context, String path) {
...
+            String providerXpathExpr = "//div[contains(@class,'credentials-add-menu-items')]"
+                    + "/div[@class='bd']/ul[@class='first-of-type']/li[contains(@class, 'yuimenuitem')]"
+                    + "/span[contains(@class,'yuimenuitemlabel') and contains(@tooltip, 'Jenkins Credentials Provider')]";
+            waitFor(by.xpath(providerXpathExpr)).click();
+        } else {
+            // Can be changed to 'find(...)' when https://github.com/jenkinsci/jenkins/pull/9835 is merged
+            all(by.css(".jenkins-dropdown"))
+                    .get(1)
+                    .findElement(by.button("Jenkins Credentials Provider"))
+                    .click();
```
[the remark this answers is not in the document]
Comment (basil) [reply src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java:50]: Never mind, I misread this.

[the remark this answers is not in the document]
Comment (janfaracik) [reply src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java:50]: Correct me if I'm wrong, would that not still be necessary for running the ATH against Jenkins versions lower than the weekly that PR will be released in? (unless that's not a thing that happens?)

[the remark this answers is not in the document]
Comment (timja) [reply src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java:50]: we only run against the latest weekly that ATH is running on (and latest LTS), (which in 99% will be latest unless something is broken and preventing us from updating)

**Comment by basil** [comment]
@janfaracik I'm getting the same `java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 1` error on 2.480 in #1771

**Pull request merged**